app/src/sharedTest/java/org/oppia/android/app/onboarding/CreateProfileFragmentTest.kt

```diff
@@ -0,0 +1,543 @@
+      testCoroutineDispatchers.runCurrent()
+
+      // No screen change as the navigation to the next screen is not implemented yet.
+      // This should fail in the future once the screen has been implemented.
+      onView(withId(R.id.create_profile_nickname_label))
+        .check(
+          matches(
+            withText(
+              context.getString(
+                R.string.create_profile_activity_nickname_label
+              )
+            )
+          )
+        )
+    }
+  }
+
+  @Test
+  fun testFragment_landscapeMode_filledNickname_continueButtonClicked_doesNotShowErrorText() {
+    launchNewLearnerProfileActivity().use {
+      onView(withId(R.id.create_profile_nickname_edittext))
+        .perform(
+          editTextInputAction.appendText("John"),
+          closeSoftKeyboard()
+        )
+      testCoroutineDispatchers.runCurrent()
+      onView(withId(R.id.onboarding_navigation_continue))
+        .perform(click())
+      testCoroutineDispatchers.runCurrent()
+
+      onView(withText(R.string.create_profile_activity_nickname_error))
+        .check(matches(withEffectiveVisibility(Visibility.GONE)))
+    }
+  }
+
+  @Test
+  fun testFragment_landscapeMode_continueButtonClicked_emptyNickname_showNicknameErrorText() {
+    launchNewLearnerProfileActivity().use {
+      onView(isRoot()).perform(orientationLandscape())
+      onView(withId(R.id.onboarding_navigation_continue))
+        .perform(click())
+      testCoroutineDispatchers.runCurrent()
+      onView(withText(R.string.create_profile_activity_nickname_error))
+        .check(matches(isDisplayed()))
+    }
+  }
+
+  @Test
+  fun testFragment_landscape_continueButtonClicked_afterErrorShown_launchesLearnerIntroScreen() {
+    launchNewLearnerProfileActivity().use {
+      onView(isRoot()).perform(orientationLandscape())
+      testCoroutineDispatchers.runCurrent()
+      onView(withId(R.id.onboarding_navigation_continue))
+        .perform(click())
+      testCoroutineDispatchers.runCurrent()
+      onView(withText(R.string.create_profile_activity_nickname_error))
+        .check(matches(isDisplayed()))
+
+      onView(withId(R.id.create_profile_nickname_edittext))
+        .perform(
+          editTextInputAction.appendText("John"),
+          closeSoftKeyboard()
+        )
+      testCoroutineDispatchers.runCurrent()
+      onView(withId(R.id.onboarding_navigation_continue))
+        .perform(click())
+      testCoroutineDispatchers.runCurrent()
+
+      // No screen change as the navigation to the next screen is not implemented yet.
+      // This should fail in the future once the screen has been implemented.
+      onView(withId(R.id.create_profile_nickname_label))
+        .check(
+          matches(
+            withText(
+              context.getString(
+                R.string.create_profile_activity_nickname_label
+              )
+            )
+          )
+        )
+    }
+  }
+
+  @RunOn(TestPlatform.ESPRESSO) // Testing lifecycle fails on Robolectric.
```

Comment:
Ditto for these tests (per my comment in your earlier PR--these can probably be tested on Robolectric by verifying Activity.isFinishing, instead).

Reply:
Done